Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -3657,7 +3662,12 @@ void LocalFrameView::scheduleScrollToAnchorAndTextFragment()
         RefPtr protectedThis = weakThis.get();
         if (!protectedThis)
             return;
-        protectedThis->scrollToAnchorAndTextFragmentNowIfNeeded();
+        if (!protectedThis->m_maintainScrollPositionAnchor) {
```

Comment:
I find this logic hard to follow. What if we just pass the relevant anchor as an argument to `scrollToAnchorAndTextFragmentNowIfNeeded()`?

Reply:
That doesn't quite work because various functions called in scrollToAnchorAndTextFragmentNowIfNeeded rely on m_maintainScrollPositionAnchor directly.